userland/Makefile

```diff
@@ -88,7 +89,7 @@ $(BUILDDIR):
 
 $(BUILDDIR)/app.elf: $(OBJS) $(LIBS) | $(BUILDDIR)
 	$(TRACE_LD)
-	$(Q)$(CC) -Wl,--gc-sections -Wl,--emit-relocs --entry=_start $(CFLAGS) $(CPPFLAGS) -T $(LINKER) -nostdlib $(OBJS) -Wl,--start-group $(LIBS) -Wl,--end-group -Wl,-Map=$(BUILDDIR)/app.Map -o $@
+	$(Q)$(CC) -Wl,-pie -Wl,--gc-sections -Wl,--emit-relocs --entry=_start $(CFLAGS) $(CPPFLAGS) -T $(LINKER) -nostdlib -Wl,--start-group $(OBJS) $(LIBS) -Wl,--end-group -Wl,-Map=$(BUILDDIR)/app.Map -o $@
```

Comment:
Honestly I'm not sure why we didn't need it before. It could be that the `-fPIC` passes through correctly w/out LTO, but there's a different path to the linker with LTO on, bit of a mystery.

Empirically, however, it's needed, as without it, the link step emits `lto1: warning: -mpic-register= is useless without -fpic` and doesn't work correctly.